llvm/lib/Target/AArch64/AArch64ISelLowering.cpp

```diff
@@ -13126,6 +13127,19 @@ AArch64TargetLowering::getRegForInlineAsmConstraint(
     return std::make_pair(unsigned(AArch64::ZT0), &AArch64::ZTRRegClass);
   }
 
+  // Clang will correctly decode the usage of register name aliases into their
+  // official names. However, other frontends like `rustc` do not. This allows
+  // users of these frontends to use the ABI names for registers in LLVM-style
+  // register constraints.
+  unsigned XRegFromAlias = StringSwitch<unsigned>(Constraint.lower())
+                               .Cases({"{x29}", "{fp}"}, AArch64::FP)
+                               .Cases({"{x30}", "{lr}"}, AArch64::LR)
+                               .Cases({"{x31}", "{sp}"}, AArch64::SP)
+                               .Default(AArch64::NoRegister);
+  if (XRegFromAlias != AArch64::NoRegister) {
+    return std::make_pair(XRegFromAlias, &AArch64::GPR64RegClass);
+  }
```

Comment:
https://llvm.org/docs/CodingStandards.html#don-t-use-braces-on-simple-single-statement-bodies-of-if-else-loop-statements